backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/DescriptorByIdSignatureFinder.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+package org.jetbrains.kotlin.backend.konan.serialization
+
+import org.jetbrains.kotlin.descriptors.*
+import org.jetbrains.kotlin.ir.util.IdSignature
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
+
+// The code here is intentionally copy-pasted from DeclarationStubGenerator with
+// minor changes.
+// "Find descriptor by IdSignature" task appears to be common and should be unified later.
+class DescriptorByIdSignatureFinder(
+        private val moduleDescriptor: ModuleDescriptor
+) {
+    fun findDescriptorBySignature(signature: IdSignature): DeclarationDescriptor? = when (signature) {
+        is IdSignature.AccessorSignature -> findDescriptorForAccessorSignature(signature)
+        is IdSignature.PublicSignature -> findDescriptorForPublicSignature(signature)
+        else -> error("only PublicSignature or AccessorSignature should reach this point, got $signature")
+    }
+
+    private fun findDescriptorForAccessorSignature(signature: IdSignature.AccessorSignature): DeclarationDescriptor? {
+        val propertyDescriptor = findDescriptorBySignature(signature.propertySignature) as? PropertyDescriptor
+                ?: return null
+        return propertyDescriptor.accessors.singleOrNull {
+            it.name == signature.accessorSignature.declarationFqn.shortName()
+        }
+    }
+
+    private fun findDescriptorForPublicSignature(signature: IdSignature.PublicSignature): DeclarationDescriptor? {
+        val packageDescriptor = moduleDescriptor.getPackage(signature.packageFqName())
+        val pathSegments = signature.declarationFqn.pathSegments()
+        val toplevelDescriptors = packageDescriptor.memberScope.getContributedDescriptors { name -> name == pathSegments.first() }
+                .filter { it.name == pathSegments.first() }
```

Comment:
Why filter for the second time?

Reply:
I tried to minimise changes to the code which I copy-pasted from `DeclarationStubGenerator` (see a commentary above). I agree that it could be rewritten. 
My plan is to upstream changes to the main Kotlin repo a bit later and present all of the changes there, so it would be easier for other users of this code to review.

Reply:
Ok.